sql/2022/pwa/assetlink_usage.sql

```diff
@@ -0,0 +1,61 @@
+#standardSQL
+# assetlink usage
+
+SELECT
+  'PWA sites' AS type,
+  _TABLE_SUFFIX AS client,
+  COUNT(0) AS freq,
+  total,
+  COUNT(0) / total AS pct
+FROM
+  `httparchive.pages.2022_07_01_*`
+JOIN
+  (
+    SELECT
+      _TABLE_SUFFIX,
+      COUNT(0) AS total
+    FROM
+      `httparchive.pages.2022_07_01_*`
+    WHERE
+      JSON_EXTRACT(payload, '$._pwa.manifests') != '[]' AND JSON_EXTRACT(payload, '$._pwa.manifests') != '{}' AND
+      JSON_EXTRACT(payload, '$._pwa.serviceWorkerHeuristic') = 'true'
+    GROUP BY
+      _TABLE_SUFFIX
+  )
+USING (_TABLE_SUFFIX)
```

Comment:
I'd love to pull these all out to CTEs to make them more readable.

```sql
WITH service_worker_totals AS (
    SELECT
      _TABLE_SUFFIX AS client,
      COUNT(0) AS total
    FROM
      `httparchive.pages.2022_07_01_*`
    WHERE
      JSON_EXTRACT(payload, '$._pwa.manifests') != '[]' AND JSON_EXTRACT(payload, '$._pwa.manifests') != '{}' AND
      JSON_EXTRACT(payload, '$._pwa.serviceWorkerHeuristic') = 'true'
    GROUP BY
      _TABLE_SUFFIX
),

totals AS (
    SELECT
      _TABLE_SUFFIX AS client,
      COUNT(0) AS total
    FROM
      `httparchive.pages.2022_06_01_*`
    GROUP BY
      _TABLE_SUFFIX
)


SELECT
  'PWA sites' AS type,
  _TABLE_SUFFIX AS client,
  COUNT(0) AS freq,
  total,
  COUNT(0) / total AS pct
FROM
  `httparchive.pages.2022_06_01_*`
JOIN
  service_worker_totals
USING (client)
WHERE
  JSON_EXTRACT(payload, '$._pwa.serviceWorkerHeuristic') = 'true' AND
  JSON_EXTRACT(payload, '$._pwa.manifests') != '[]' AND JSON_EXTRACT(payload, '$._pwa.manifests') != '{}' AND
  JSON_EXTRACT_SCALAR(JSON_VALUE(payload, '$._well-known'), "$['/.well-known/assetlinks.json'].found") = 'true'
GROUP BY
  client,
  total
UNION ALL
SELECT
  'All sites' AS type,
  _TABLE_SUFFIX AS client,
  COUNT(0) AS freq,
  total,
  COUNT(0) / total AS pct
FROM
  `httparchive.pages.2022_06_01_*`
JOIN
  totals
USING (client)
WHERE
  JSON_EXTRACT_SCALAR(JSON_VALUE(payload, '$._well-known'), "$['/.well-known/assetlinks.json'].found") = 'true'
GROUP BY
  client,
  total
ORDER BY
  type DESC,
  freq / total DESC,
  client
```

This is actually cleaning up the SQL from last year (which I had a hand in!) rather than anything you did wrong and it's a bit of work to restructure all the queries in this way, so understand you may be reluctant to take this on, but think it would be cleaner going forward. WDYT?

Reply:
Not sure if I have enough time to get this done before month ends. If the refactor can come in a later date, I am happy to take a stab at them. Thoughts?

Reply:
Yeah that's fine. Let's merge this as is.

I think it's debatably if it's that much clearer. But certain with more embedded subqueries it would get unreadable so probably about the limit of what we'd want now. Let's leave as is.